Updates to Introduction to Clarify Purpose of the Specification

rstreif opened the pull request:
Added clarification to the introduction section to explain what this
specification is intended for and what it is not. This is to address concerns voiced by the Mozilla team on [https://groups.google.com/forum/#!topic/mozilla.dev.platform/EtMiqNFswq0].

Signed-off-by: Rudolf J Streif <rstreif@jaguarlandrover.com>

wonsuk73 commented:
@rstreif Thanks for proposal! One thing I need to check is whether is it for abstract part or introduction part?. You changed abstract part but your commented mention introduction :)  To me abstract is better. So I think some of sentences might be moved to introduction parts. What do you think? As a reference it was discussed in #109

PeterWinzell commented:
Minor correction

...Client applications can be web applications that run inside a browser or an embedded web runtime, or they are application**S**

@rstreif and @wonsuk73  I think that maybe we should shorten this section somewhat:

"This is a departure from the typical web application and web IDL model where APIs are implemented as browser or web runtime extensions. Even if the type of applications  **WERE** limited to web applications the web IDL approach would mean for the APIs to be implemented for many different embedded browsers and web runtime envrionments used by the automotive industry for their platforms. That would rather uneconomical and hinder adoption.Furthermore, access to the vehicle signal server is restricted to local clients, meaning clients running on the same host or on a local network, only. The vehicle signal server is not intended to expose its web socket interface to external networks or the Internet. For such architecture to function correctly additional services are required such as but not limited to **discovery and resolution.**"

Something like: This departure from the traditional Web IDl model is done to promote adaption and to avoid browser implementation dependency. The vehicle signal server is not intended to expose its web socket interface to external networks, thus, client access is locally restricted. 

In general I feel that it is a bit strange that we in a specification of an API need to explain how the automotive industry work. However, in the abstract as @wonsuk73 says it might be the right thing to do.

drkevg committed:
Removed duplicate statement defining WebSocket API from Intro

acrofts84 committed:
Merge pull request #112 from w3c/drkevg-patch-2

Removed duplicate statement defining WebSocket API from Intro

rstreif commented:
Hi Wonsuk,

@rstreif https://github.com/rstreif Thanks for proposal! One think I need

> to check is whether is it for abstract part or introduction part?. You
> changed abstract part but your comment mention introduction :) To me
> abstract is better. So I think some of sentences might be moved to
> introduction parts. What do you think? As a reference it was discussed in
> #109 https://github.com/w3c/automotive/pull/109
> 
> Yes, it was meant to be abstract and not introduction.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/w3c/automotive/pull/111#issuecomment-260224810, or mute
> the thread
> https://github.com/notifications/unsubscribe-auth/ALBtqD0r3z2iCf0ZMD18BrrvdWs9TTRNks5q96sdgaJpZM4KvRMR
> .

## 

_Rudolf J Streif_
System Architect - Open Source Initiative
Open Source Technology Centre

_M:_ +1.619.631.5383
_Email:_  rstreif@jaguarlandrover.com

UK: G/26/2 G02 Building 523, Engineering Centre, Gaydon, Warwick, CV35 ORR
US: 1419 NW 14th Ave, Portland, OR 97209

## jaguar.com | landrover.com

Business Details:
Jaguar Land Rover Limited
Registered Office: Abbey Road, Whitley, Coventry CV3 4LF
Registered in England No: 1672070

This e-mail and any attachments contain confidential information for a
specific individual and purpose.  The information is private and privileged
and intended solely for the use of the individual to whom it is addressed.
If you are not the intended recipient, please e-mail us immediately.  We
apologise for any inconvenience caused but you are hereby notified that any
disclosure, copying or distribution or the taking of any action in reliance
on the information contained herein is strictly prohibited.

This e-mail does not constitute an order for goods or services unless
accompanied by an official purchase order.

rstreif commented:
@PeterWinzell I am ok with shortening it. The important parts are promoting adoption, avoiding browser dependencies and restricted to access from local networks, which is what you covered.

drkevg committed:
Update title to Vehicle Information Service Spec

drkevg committed:
Merge pull request #114 from w3c/drkevg-title

Update title to Vehicle Information Service Spec

drkevg committed:
fixed typo and missed title change

drkevg committed:
Component diagram with new spec titles

drkevg committed:
Deleted new component image file prior to renaming and uploading

drkevg committed:
Uploaded component diagram with new spec. titles

drkevg committed:
Renaming

drkevg committed:
Renamed component diagram with new titles

drkevg committed:
Updated component diagram to include new spec titles

acrofts84 committed:
Use DOMString instead of Strings

Ensure all strings are DOMStrings i.a.w. WebIDL and to ensure the document is consistent

acrofts84 committed:
Note to state that JSON is the data format used throughout

"The data format used for client/server communication within this specification is JSON encoded strings, however other data formats may be supported in future versions of the specification."

acrofts84 committed:
WebSocket security token diagram clarification

The diagram in Figure 3 does not use the exact data structures from the specification, but it is demonstrative and simpler than including the actual structures which would need additional fields, such as timestamps and requestIds. For simplicity I think the diagram should remain the same, but it is important to note that it is conceptual.

acrofts84 committed:
Clarify behaviour for unsubscribeAll action

If the unsubscribeAll action is used, but there are no active subscriptions, this will still return a success response as the subscriptions will be in the state desired by the user.

acrofts84 committed:
Clarify behaviour of subscriptions when authentication fails

"If the authentication token expires whilst a subscription is active, the server will send a subscriptionNotificationError. If authentication is re-established with the server the subscription must continue the subscription without requiring another subscription request."

acrofts84 committed:
Additional Errors - read_only and too_many_attempts

Added read_only as an error for setting signals.
Added too_many_attempts to prevent dictionary attacks for authentication

Commit:
Add Vehicle Signal Client Spec as markdown

acrofts84 committed:
Place error after requestId for consistency

acrofts84 committed:
Add vss request and response objects to the Message Structure table

acrofts84 committed:
Working Document for JSON Schema Note to Complement VISS

acrofts84 committed:
Update Name

wonsuk73 committed:
Merge pull request #122 from pkinney/patch-1

Add Vehicle Signal Client Spec

acrofts84 committed:
Minor update to text

acrofts84 committed:
Merge pull request #120 from w3c/acrofts84-errors

Additional Errors - read_only and too_many_attempts

acrofts84 committed:
Satisfy all cases for subscription authentication

acrofts84 committed:
Merge pull request #119 from w3c/acrofts84-patch-6

Discussed and agreed with @drkevg

acrofts84 committed:
Merge branch 'gh-pages' into acrofts84-patch-2

acrofts84 committed:
Merge pull request #115 from w3c/acrofts84-patch-2

Use DOMString instead of Strings

acrofts84 committed:
Merge pull request #118 from w3c/acrofts84-patch-5

Clarify behaviour for unsubscribeAll action

acrofts84 committed:
Semantic change

acrofts84 committed:
Merge pull request #117 from w3c/acrofts84-patch-4

WebSocket security token diagram clarification

acrofts84 committed:
Merge pull request #116 from w3c/acrofts84-patch-3

Note to state that JSON is the data format used throughout

acrofts84 committed:
Minor change to subscriptionNotificationError comment

acrofts84 committed:
Text update

acrofts84 committed:
Update VISS_JSON_Schema_Note.html

acrofts84 committed:
requestId and subscriptionId should be strings

This allows them to be numeric or UUIDs etc.

acrofts84 committed:
Note on wwwivi as a hostname

Discuss merits of using 'wwwivi' as a standardised hostname. Do we want a more generic hostname, not just for IVI but also encompassing WoT standards?

wonsuk73 committed:
Merge pull request #128 from w3c/acrofts84-reqIds

requestId and subscriptionId should be strings

wonsuk73 committed:
Merge pull request #129 from w3c/acrofts84-patch-1

Note on wwwivi as a hostname

acrofts84 committed:
Update filters schema

acrofts84 committed:
Update to filters schema

acrofts84 committed:
Update VISS_JSON_Schema_Note.html

acrofts84 committed:
Add JSON Schema Definitions to Each Interface

acrofts84 committed:
Clarify the getVSS method

Introduce a data structure in the example and add further explanation

acrofts84 committed:
Update request signal value

From Signal.Body to Signal.Drivetrain.InternalCombustionEngine.RPM

acrofts84 committed:
Clarify Server Side Filter Behaviour with JSON Schema

acrofts84 committed:
Neutral wording and typo in server side filter

wonsuk73 committed:
Merge pull request #133 from w3c/acrofts84-getVSS

Clarify the getVSS method

acrofts84 committed:
Provide a more substantial example

Added getVSS, subscriptionIds, and a generic onMessage handler

acrofts84 committed:
Clarify subscription behaviour

Explicitly state that the tree unsubscribes from an entire subscription, you cannot just do just a subset

marcelbuesing committed:
Remove redundant object in Set example

Fix invalid JSON in Set example, resolve #150

Commit:
Merge pull request #152 from marcelbuesing/gh-pages

Remove redundant object in Set example

acrofts84 committed:
Merge branch 'gh-pages' into acrofts84-json-schema

acrofts84 committed:
Merge pull request #131 from w3c/acrofts84-json-schema

Add JSON Schema Definitions to Each Interface

acrofts84 committed:
Add subscriptionHandler and subscriptionSetupHandler to account for the "subscription" action

acrofts84 committed:
Add "subscription" action

acrofts84 committed:
Merge pull request #153 from w3c/acrofts84-example

Integrate new example

acrofts84 committed:
Minor textual update

acrofts84 committed:
Removed repetition from the introduction:
"RESTful web services are out of scope for the first revision of this specification, but could be considered for addition in a later version."

"For security reasons, clients are not able to connect directly to ECUs or to CAN, MOST or LIN networks. All access is via WebSocket. This allows the server to securely control access to vehicle signals."

Removed repetition from "Security and Privacy Considerations - Use of Encryption"

"To support ‘defence in depth’ and a layered security approach, connections between clients and servers are strongly encrypted. This is to make it more difficult for an attacker that has succeeded in installing malicious code on a vehicle to eavesdrop, hijack security tokens or impersonate valid security principals in order to get and set sensitive vehicle signals."

acrofts84 committed:
Merge pull request #154 from w3c/acrofts84-subscriptionAction

Add "subscription" action and update example

wonsuk73 committed:
Merge pull request #139 from w3c/acrofts84-unsubscribe

Clarify subscription behaviour

wonsuk73 committed:
Merge pull request #155 from w3c/acrofts84-initSection

Removed repetition in the Initialisation section
